### ISSUE TITLE
`addon.tab.editorMode` should not rely on the URL

### ISSUE THREAD

**Issue by WorldLanguages** [ScratchAddons/ScratchAddons#6452]
### Description

`addon.tab.editorMode` always returns `editor` if the URL does not change while on fullscreen.

**This is not the case when running Scratch Addons (in dev mode) on scratchfoundation.github.io as there's a special if statement for that that uses Redux to check for fullscreen.**

### Steps to reproduce

1. Go to https://scratch.mit.edu/projects/editor/ while logged out
2. Go full screen
3.  `addon.tab.editorMode` still returns `editor`

### Demonstration(s) and additional context

The easiest fix is probably to use Redux

### Environment

Any browser

**Comment by WorldLanguages** [comment]
Related: #6437 which suggests an `editorModeChange` event.

**Comment by WorldLanguages** [comment]
This bug may manifest itself through issues like this, when logged out:

(screenshot is fullscreen mode with debugger addon enabled)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/17484114/e439ad2c-22fb-4517-864b-a71494b0d418)

**Comment by WorldLanguages** [comment]
I believe we've had this bug for over three years.
![image](https://github.com/ScratchAddons/ScratchAddons/assets/17484114/90859a02-1dc7-4fcf-8950-b8950640f8cb)

**Comment by WorldLanguages** [comment]
#7777 did not change the behavior of `.sa-body-editor` which probably means some of the bugs I've sent here still happen.

Also, Samq said on Discord:

> Sometimes .sa-body-editor is added late which causes some weird styling issues on the editor loader for 1/4 of a second which is particularly noticeable on /projects/editor/

**Comment by mxmou** [comment]
> [#7777](https://github.com/ScratchAddons/ScratchAddons/pull/7777) did not change the behavior of `.sa-body-editor` which probably means some of the bugs I've sent here still happen.

@WorldLanguages Do you remember what you meant by this? `.sa-body-editor` handles both /projects/editor and scratchfoundation.github.io correctly.

**Comment by WorldLanguages** [comment]
You're right, I think I had a misconception on whether sa-body-editor was added on fullscreen.

**Comment by mxmou** [comment]
I'll close the issue then.
